semmc/src/SemMC/Architecture/Internal.hs

```diff
@@ -1,5 +1,6 @@
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeFamilyDependencies #-}
```

Comment:
Isn't this extension just for adding functional dependencies to a type family?

Reply:
Without it, GHC 8.10 fails with:

    src/SemMC/Architecture/Internal.hs:31:1: error:
        * Illegal result type variable (r :: (Symbol -> Type) -> [Symbol] -> Type) for 'Opcode'
            Enable TypeFamilyDependencies to allow result variable names
       |
    31 | type family Opcode (arch :: Type) = (r :: (Symbol -> Type) -> [Symbol] -> Type)
       |